<?php 
include 'header.html'; 
include 'menu.html'; 
?>
<html>
<body>

<section class="span12">  
<h1 class="page-header">About Us</h1>
<header>
</header>

<div class="field field-name-body field-type-text-with-summary field-label-hidden text-justify"><div class="field-items"><div class="field-item even" property="content:encoded">

<h3> Background</h3>
<p>An important feature of plant stress response comprises the production of peptides with e.g signaling, antimicrobial and antioxidant activities. 
The number of peptide-encoding genes annotated in the <i>A. thaliana</i> genome is still very low despite the fact that there have been 
numerous published reports on peptides encoded by sORFs in other genomes. Our current research at Centre of Microbial and Plant Genetics, KU Leuven
 focuses on the discovery of novel peptides to enrich the existing inventory of plant peptides and also to decipher the functional roles these peptides 
might exhibit as a part of stress tolerance mechanisms. Using whole genome Tiling arrays and RNA sequencing, we have identified several unannotated <b><code>transcriptionally active regions (TARs)</b></code>upon application of biotic <i>(Botrytis cinerea)</i> and abiotic stress (oxidative stress triggered by the herbicide Paraquat) in <i>A. thaliana</i>. 
These TARs could be translated into putative <b><i><code>Botrytis cinerea</i>induced peptides (BIPs)</code></b> and <b><code>Oxidative stress induced peptides (OSIPs)</code></b>respectively.<p>

<p>ARA-PEPs was developed within the Computational Systems Biology (CSB) group, Centre Of Microbial and Plant Genetics, KU Leuven. </p>

<h3>FAQs:</h3>
<p><b>What is ARA-PEPs?</b>  ARA-PEPs is a database of putative peptides encoded by sORFs in <i>A. thaliana</i> predicted from whole transcriptome analysis. 
The webserver provides a common platform to browse and access all publicly available putative peptides encoded by sORFs in the <i>A. thaliana</i> genome.<p>

<p><b>Which type of peptides can be searched?</b> Peptides encoded by sORFs with transcriptional evidence can be searched.</p>
 
<p><b>What type of information is stored in ARA-PEPs?</b> ARA-PEPs stores experimentally studied transcriptionally active regions (TARs) from in-house and publicly available 
Tiling Array, RNA-seq experiments and predicted based on bioinformatics analysis.</p>
 
<p><b>How to browse and download data from ARA-PEPs?</b> ARA-PEPs has a very easy user interface. By clicking on the <code>Datasets</code> tab, one can search the independent datasets. Users can customize their query by clicking on the <code>SEARCH peptides</code> tool. Users can also download information from the query results by exporting a csv file or copying to clipboard or by accessing the download tab in the toolbar. Detailed method of browsing and downloading information from the database is explained in the <code>Help</code> section of the webserver.</p>

</ul>  
</body>
</html>


<?php include 'footer.html'; ?>

